server/src/domain/storage/storage.core.ts

```diff
@@ -140,29 +143,74 @@ export class StorageCore {
 
     let move = await this.moveRepository.getByEntity(entityId, pathType);
     if (move) {
+      console.log('Found existing move record for', entityId, pathType, oldPath, newPath);
       this.logger.log(`Attempting to finish incomplete move: ${move.oldPath} => ${move.newPath}`);
       const oldPathExists = await this.repository.checkFileExists(move.oldPath);
       const newPathExists = await this.repository.checkFileExists(move.newPath);
-      const actualPath = newPathExists ? move.newPath : oldPathExists ? move.oldPath : null;
+      const actualPath = oldPathExists ? move.oldPath : newPathExists ? move.newPath : null;
       if (!actualPath) {
         this.logger.warn('Unable to complete move. File does not exist at either location.');
         return;
       }
 
-      this.logger.log(`Found file at ${actualPath === move.oldPath ? 'old' : 'new'} location`);
+      const fileAtNewLocation = actualPath === move.newPath;
+      this.logger.log(`Found file at ${fileAtNewLocation ? 'new' : 'old'} location`);
+
+      if (fileAtNewLocation) {
+        if (!(await this.isNewPathSameSize(move.oldPath, move.newPath, assetInfo))) {
+          this.logger.fatal(`Skipping move due to file size mismatch, old file is missing and new file is different`);
+          return;
+        }
+      }
 
       move = await this.moveRepository.update({ id: move.id, oldPath: actualPath, newPath });
     } else {
       move = await this.moveRepository.create({ entityId, pathType, oldPath, newPath });
     }
```

Comment:
We should probably flip this to decrease the overall indentation. So `if (!move) { ... } else { ... }`

Reply:
Btw nvm this. This would only apply if we were to move all of this to a separate method and early return from it